git_of_theseus/analyze.py

```diff
@@ -21,37 +21,46 @@
 IGNORE_PYGMENTS_FILETYPES = ['*.json', '*.md', '*.ps', '*.eps', '*.txt', '*.xml', '*.xsl', '*.rss', '*.xslt', '*.xsd', '*.wsdl', '*.wsf', '*.yaml', '*.yml']
 
 
-def analyze():
+def analyze(cohortfm=None, interval=None, ignore=None, only=None, outdir=None, branch=None, all_filetypes=None, repos=None):
     default_filetypes = set()
     for _, _, filetypes, _ in pygments.lexers.get_all_lexers():
         default_filetypes.update(filetypes)
     default_filetypes.difference_update(IGNORE_PYGMENTS_FILETYPES)
 
     parser = argparse.ArgumentParser(description='Analyze git repo')
```

Comment:
can you move the argument parser out of this function? can put it under `if __name__ == '__main__'` or in a separate function.